java-util/src/main/java/io/druid/java/util/common/parsers/AbstractFlatTextFormatParser.java

```diff
@@ -65,13 +66,39 @@ public String getDefaultDelimiter()
 
   public AbstractFlatTextFormatParser(
       @Nullable final String listDelimiter,
+      @Nullable final Map<String, String> multiValueDelimiter,
       final boolean hasHeaderRow,
       final int maxSkipHeaderRows
   )
   {
     this.listDelimiter = listDelimiter != null ? listDelimiter : Parsers.DEFAULT_LIST_DELIMITER;
-    this.listSplitter = Splitter.on(this.listDelimiter);
-    this.valueFunction = ParserUtils.getMultiValueFunction(this.listDelimiter, this.listSplitter);
+    Map<String, Splitter> deliSpliterMap = new HashMap<>();
+    this.multiValueTransformer = new Maps.EntryTransformer<String, String, Object>()
+    {
+      @Override
+      public Object transformEntry(String key, String value)
+      {
+        String deli = null;
+        if (multiValueDelimiter == null) {
+          deli = AbstractFlatTextFormatParser.this.listDelimiter;
+        } else if (multiValueDelimiter.get(key) != null) {
+          deli = multiValueDelimiter.get(key);
+        }
+
+        if (deli != null && value.contains(deli)) {
+          Splitter multiSplitter = deliSpliterMap.get(deli);
+          if (multiSplitter == null) {
+            multiSplitter = Splitter.on(deli);
+            deliSpliterMap.put(deli, multiSplitter);
+          }
+          return StreamSupport.stream(multiSplitter.split(value).spliterator(), false)
+                  .map(Strings::emptyToNull)
```

Comment:
Why emptyToNull?

Reply:
Just follow original logic, transform empty to null